src/Microsoft.Health.Dicom.Functions/Indexing/ReindexOperation.cs

```diff
@@ -0,0 +1,141 @@
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Indexing;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+using Microsoft.Health.Dicom.Functions.Indexing.Models;
+
+namespace Microsoft.Health.Dicom.Functions.Indexing
+{
+    /// <summary>
+    /// Represents the Azure Durable Functions that perform the re-indexing of previously added DICOM instances
+    /// based on new tags configured by the user.
+    /// </summary>
+    public partial class ReindexOperation
+    {
+        private readonly ReindexConfiguration _reindexConfig;
+        private readonly ITagReindexOperationStore _tagOperationStore;
+        private readonly IInstanceReindexer _instanceReindexer;
+        private readonly IAddExtendedQueryTagService _addExtendedQueryTagService;
+
+        public ReindexOperation(IOptions<IndexingConfiguration> configOptions,
+            IAddExtendedQueryTagService addExtendedQueryTagService,
+            ITagReindexOperationStore tagOperationStore,
+            IInstanceReindexer instanceReindexer)
+        {
+            EnsureArg.IsNotNull(configOptions, nameof(configOptions));
+            EnsureArg.IsNotNull(tagOperationStore, nameof(tagOperationStore));
+            EnsureArg.IsNotNull(instanceReindexer, nameof(instanceReindexer));
+            EnsureArg.IsNotNull(addExtendedQueryTagService, nameof(addExtendedQueryTagService));
+            _reindexConfig = configOptions.Value.Add;
+            _tagOperationStore = tagOperationStore;
+            _instanceReindexer = instanceReindexer;
+            _addExtendedQueryTagService = addExtendedQueryTagService;
+        }
+
+        /// <summary>
+        ///  The orchestration function to add extended query tags.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>The task.</returns>
+        [FunctionName(nameof(AddExtendedQueryTagsOrchestrationAsync))]
+        public async Task AddExtendedQueryTagsOrchestrationAsync(
+            [OrchestrationTrigger] IDurableOrchestrationContext context,
+            ILogger logger)
+        {
+            EnsureArg.IsNotNull(context, nameof(context));
+            logger = context.CreateReplaySafeLogger(logger);
+            var input = context.GetInput<IEnumerable<AddExtendedQueryTagEntry>>();
+            var tagEntries = await context.CallActivityAsync<IEnumerable<ExtendedQueryTagStoreEntry>>(nameof(AddExtendedQueryTagAsync), input);
```

Comment:
What if the tag already exists and it cannot be added? I would think we'd just stop processing immediately with a "Failed" status

Reply:
yes, that's right. eh, problem with it?

Reply:
I was just wondering because it doesn't appear to be gracefully handled. Ideally we'd set a custom status such that it can be easily seen when a user queries the status of an operation.